<launch>
    <arg name="save_map" default="false"/>
    <arg name="load_map" default="false"/>
    <arg name="map_only" default="false"/>
    <arg name="localization_mode" default="false"/>
    <arg name="sensor_type" default="imu-monocular"/>
    <arg name="use_pangolin" default="true"/>
    <arg name="use_live_feed" default="true"/>
    <arg name="video_name" default="default.mp4"/>
    <arg name="playback_bag" default="changeme"/>
    <arg name="record_bag" default="changeme"/>
    <arg name="bag_name" default="$(env USER)_$(eval 'datetime.datetime.now().strftime(\'%Y-%m-%d_%H-%M-%S\')')"
    description="An optional argument for specifying a custom bag name. Otherwise, the date and time will be used for the name of the bag file."/>
    <include file="$(find-pkg-share realsense2_camera)/launch/rs_launch.py" if="$(eval '\'$(var use_live_feed)\' == \'true\'')">
        <arg name="initial_reset" value="true"/>
        <arg name="rgb_camera.color_profile" value="640x480x30"/>
        <arg name="enable_accel" value="true"/>
        <arg name="enable_gyro" value="true"/>
        <arg name="unite_imu_method" value="2"/>
        <arg name="enable_depth" value="false"/>
    </include>
    <node pkg="orb_slam3_ros2" exec="imu_mono_node_cpp" output="screen">
        <param name="map_only" value="$(var map_only)"/>
        <param name="localization_mode" value="$(var localization_mode)"/>
        <param name="sensor_type" value="$(var sensor_type)"/>
        <param name="use_pangolin" value="$(var use_pangolin)"/>
        <param name="use_live_feed" value="$(var use_live_feed)"/>
        <param name="video_name" value="$(var video_name)"/>
    </node>
    <executable cmd="ros2 bag play $(find-pkg-share orb_slam3_ros2)/bags/$(var playback_bag)" if="$(eval '\'$(var playback_bag)\' != \'changeme\'')"/>
    <executable cmd="ros2 bag record -a -o ./ORB_SLAM3_ROS2/bags/$(var bag_name)" if="$(eval '\'$(var record_bag)\' != \'true\'')"/>
</launch>
